contracts/CurrencyNetwork.sol

```diff
@@ -448,6 +496,18 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
             forwardedValue += fee;
             fees += fee;
             require(fees <= _maxFee);
+
+            if (safeInterestRippling) {
+                // we want to prevent intermediaries to pay more interests than they receive
+                // If the transaction does not result in them paying interests but getting paid less interests then transaction is valid
+                // i.e. transaction is valid if the receiver still owes the sender after transfer
+                Account memory account = _loadAccount(sender, receiver);
```

Comment:
Yes, this is indeed a problem.
Originally, I had _directTransfer return an account to _mediatedTransfer, to avoid having to load it twice (because I thought it was gas intensive), which meant I did not have this problem, but it seems I was wrong and thought it was less readable.
The solutions is either to go back to this idea of loading only one account and passing it on, or I modify and store the account at the end of _mediatedTransfer.
Whichever you like.